src/mlmc/quantity_concept.py

```diff
@@ -62,28 +189,39 @@ def __init__(self, input_quantities, operation = None):
         # self._chunk = None
         # #  memoized  resulting chunk
 
+    def set_range(self, start, size):
+        for quantity in self._input_quantities:
+            quantity.set_range(start, size)
+
+    def size(self) -> int:
+        return self.qtype.size()
+
+    def get_cache_key(self, level_id, i_chunk):
+        return str(hash((level_id, i_chunk, id(self), *[id(q) for q in self._input_quantities])))
 
-    def _samples(self, i_chunk):
+    @clearable_cache(custom_key_maker=get_cache_key)
```

Comment:
? why clearable
? can we use max_size=1
? simplification of get_cache_key:
  - just return tuple
  - no need to hash also input quantities as these are unique for self anyway